--- /var/www/zencart-pro6pp/includes/form_check.js.php	2012-09-18 21:36:42.000000000 +0200
+++ /home/nionios/d-centralize/workspace/zenCart/modified_zencart_pro6pp/modify_files/includes/form_check.js.php	2013-11-29 11:06:38.401378577 +0100
@@ -6,7 +6,7 @@
  * @copyright Copyright 2003-2010 Zen Cart Development Team
  * @copyright Portions Copyright 2003 osCommerce
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id: form_check.js.php 17018 2010-07-27 07:25:41Z drbyte $
+ * @version $Id: form_check.js.php 16186 2010-05-03 18:38:57Z drbyte $
  */
 ?>
 <script language="javascript" type="text/javascript"><!--
@@ -145,6 +145,7 @@ function check_form(form_name) {
 <?php } ?>
 <?php if ((int)ENTRY_STREET_ADDRESS_MIN_LENGTH > 0) { ?>
   check_input("street_address", <?php echo (int)ENTRY_STREET_ADDRESS_MIN_LENGTH; ?>, "<?php echo ENTRY_STREET_ADDRESS_ERROR; ?>");
+  check_input("street_no", <?php echo ENTRY_STREET_NO_MIN_LENGTH; ?>, "<?php echo ENTRY_STREET_NO_ERROR; ?>");  
 <?php } ?>
 <?php if ((int)ENTRY_POSTCODE_MIN_LENGTH > 0) { ?>
   check_input("postcode", <?php echo (int)ENTRY_POSTCODE_MIN_LENGTH; ?>, "<?php echo ENTRY_POST_CODE_ERROR; ?>");
@@ -153,7 +154,7 @@ function check_form(form_name) {
   check_input("city", <?php echo (int)ENTRY_CITY_MIN_LENGTH; ?>, "<?php echo ENTRY_CITY_ERROR; ?>");
 <?php } ?>
 <?php if (ACCOUNT_STATE == 'true') { ?>
-  check_state(<?php echo (int)ENTRY_STATE_MIN_LENGTH . ', "' . ENTRY_STATE_ERROR . '", "' . ENTRY_STATE_ERROR_SELECT; ?>");
+  check_state(<?php echo (int)ENTRY_STATE_MIN_LENGTH . ', "' . ENTRY_STATE_ERROR . '", "' . ENTRY_STATE_ERROR_SELECT; ?>);
 <?php } ?>
 
   check_select("country", "", "<?php echo ENTRY_COUNTRY_ERROR; ?>");
@@ -175,4 +176,4 @@ function check_form(form_name) {
     return true;
   }
 }
-//--></script>
+//--></script>
